Lib/http/server.py

```diff
@@ -1051,9 +1052,22 @@ def finish_request(self, request, client_address):
             self.RequestHandlerClass(request, client_address, self,
                                      directory=args.directory)
 
+    class HTTPSDualStackServer(ThreadingHTTPSServer):
```

Comment:
Instead of duplicating the class, I suggest using a mixin class:

```py
class DualStackServerMixin:
        def server_bind(self):
            # suppress exception when protocol is IPv6
            with contextlib.suppress(Exception):
                self.socket.setsockopt(
                    socket.IPPROTO_IPV6, socket.IPV6_V6ONLY, 0)
            return super().server_bind()

        def finish_request(self, request, client_address):
            self.RequestHandlerClass(request, client_address, self,
                                     directory=args.directory)

class HTTPDualStackServer(DualStackServerMixin, ThreadingHTTPServer): pass
class HTTPSDualStackServer(DualStackServerMixin, ThreadingHTTPSServer): pass
```